### PR TITLE
[Testing] Updated the contents about getting the container in tests

### PR DESCRIPTION
**Pull request by javiereguiluz**
I know that using the container in tests is an edge-case ... but still, I think that the current docs are confusing and convoluted. This PR proposes to simplify the explanation of the different container types.

### CODE REVIEW

**Review by wouterj — APPROVED** [review]

**Comment by wouterj** [comment]
Thank you Javier. I think this is much clearer indeed :)

File: testing.rst [hunk 1]
```diff
@@ -544,74 +544,51 @@ You can also get the objects related to the latest request::
...
-            // ...
+Finally, for the most rare edge-cases, Symfony includes a special container
+which provides access to all services, public and private. This special
+container is a service that can be get via the normal container::
 
-            $services->alias('test.App\Test\SomeTestHelper', 'App\Test\SomeTestHelper')->public();
-        };
+    public function testSomething()
+    {
+        $client = self::createClient();
```
Comment (alexisLefebvre) [inline testing.rst:584]: `createClient` doesn't exist when a test class inherit from `KernelTestCase`. Is there a workaround?
